spec/policies/agent/agent_role_policy_spec.rb

```diff
@@ -1,68 +1,56 @@
-      let!(:agent_role) { create(:agent_role, access_level: AgentRole::ACCESS_LEVEL_ADMIN, agent: other_agent, organisation: other_organisation) }
+    context "admin access to this territory" do
+      let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: [territory]) }
+      let!(:agent_role) { create(:agent_role, organisation: organisation) }
 
-      permissions(:update?) { it { is_expected.not_to permit(pundit_context, agent_role) } }
+      it_behaves_like "permit actions", :update?, :edit?, :create?, :destroy?
     end
-  end
-end
 
-RSpec.describe Agent::AgentRolePolicy::Scope, type: :policy do
-  describe "#resolve?" do
-    subject { described_class.new(AgentContext.new(agent), AgentRole).resolve }
+    context "admin access to a different territory" do
+      let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: [territory]) }
+      let!(:agent_role) { create(:agent_role, organisation: create(:organisation)) }
 
-    context "misc state" do
-      let!(:organisations) { create_list(:organisation, 4) }
-      let!(:agent) { create(:agent) }
-      let!(:own_agent_role_basic) { create(:agent_role, agent: agent, organisation: organisations[0]) }
-      let!(:own_agent_role_admin1) { create(:agent_role, :admin, agent: agent, organisation: organisations[1]) }
-      let!(:own_agent_role_admin2) { create(:agent_role, :admin, agent: agent, organisation: organisations[2]) }
-      let!(:agent_role_basic_role) { create(:agent_role, agent: create(:agent), organisation: organisations[0]) }
-      let!(:agent_role_admin_role1) { create(:agent_role, agent: create(:agent), organisation: organisations[1]) }
-      let!(:agent_role_admin_role2) { create(:agent_role, agent: create(:agent), organisation: organisations[2]) }
-      let!(:agent_role_other_orga) { create(:agent_role, agent: create(:agent), organisation: organisations[3]) }
+      it_behaves_like "not permit actions", :update?, :edit?, :create?, :destroy?
+    end
 
-      it do
-        expect(subject).to include(own_agent_role_basic)
-        expect(subject).to include(own_agent_role_admin1)
-        expect(subject).to include(own_agent_role_admin2)
+    context "allowed to invite agents access right" do
+      let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: []) }
+      let!(:agent_role) { create(:agent_role, organisation: organisation) }
 
-        expect(subject).not_to include(agent_role_basic_role)
-        expect(subject).to include(agent_role_admin_role1)
-        expect(subject).to include(agent_role_admin_role2)
-        expect(subject).not_to include(agent_role_other_orga)
+      before do
+        create(:agent_territorial_access_right, agent: agent, territory: organisation.territory, allow_to_invite_agents: true)
       end
+
+      it_behaves_like "permit actions", :update?, :edit?, :create?, :destroy?
     end
```

Comment:
si on veut faire ceinture et bretelles on peut rajouter une spec dans le cas du droit d’invitation d’agents dans un autre territoire

```suggestion

    context "allowed to invite agents access right in a different territory" do
      let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: []) }
      let!(:agent_role) { create(:agent_role, organisation: organisation) }

      before do
        create(:agent_territorial_access_right, agent: agent, territory: create(:territory), allow_to_invite_agents: true)
      end

      it_behaves_like "not permit actions", :agent_role, :update?, :edit?, :create?, :destroy?
    end
  end
```